### PR TITLE
Feature/96 improve domain and repository layers

### PR DESCRIPTION
**Pull request by NightOfNights**
#96

### CODE REVIEW

File: src/api/services/releasesService.ts [hunk 1]
```diff
@@ -23,7 +23,10 @@ interface ReleaseIcon {
   icon: string;
```
Comment (moetazaneta) [inline src/api/services/releasesService.ts]: Consider adding `readonly`

File: src/models/releases/releaseAnimation.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  /**
+   * Story scenes animation.
+   */
+  storyAnimation: ReleaseAnimationType | null;
+
+  /**
+   * Ero scenes animation.
+   */
+  eroAnimation: ReleaseAnimationType | null;
```
Comment (moetazaneta) [inline src/models/releases/releaseAnimation.ts]: Is it intended for release related models to not have `readonly`?

Comment (NightOfNights) [reply src/models/releases/releaseAnimation.ts]: @Moltenship ?

Comment (Moltenship) [reply src/models/releases/releaseAnimation.ts]: Uh, no

**Review by Moltenship — APPROVED** [review]

**Pull request merged**